internal/service/rds/custom_db_engine_version.go

```diff
@@ -0,0 +1,469 @@
+			},
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"database_installation_files_s3_bucket_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(3, 63),
+			},
+			"database_installation_files_s3_prefix": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 255),
+			},
+			"db_parameter_group_family": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"engine": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexache.MustCompile(fmt.Sprintf(`^%s.*$`, InstanceEngineCustomPrefix)), fmt.Sprintf("must begin with %s", InstanceEngineCustomPrefix)),
+					validation.StringLenBetween(1, 35),
+				),
+			},
+			"engine_version": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 60),
+			},
+			"filename": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"manifest"},
+			},
+			//API returns created image_id of the newly created image.
+			"image_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"kms_key_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"major_engine_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"manifest": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringIsJSON,
+					validation.StringLenBetween(1, 100000),
+				),
+				ConflictsWith:    []string{"filename"},
+				DiffSuppressFunc: verify.SuppressEquivalentJSONDiffs,
+				StateFunc: func(v interface{}) string {
+					json, _ := structure.NormalizeJsonString(v)
+					return json
+				},
+			},
+			//API returns manifest with service added additions, non-determinestic.
+			"manifest_computed": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"manifest_hash": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice(rds.CustomEngineVersionStatus_Values(), false),
+			},
+			// Allow CEV creation from a source AMI ID.
+			// implicit state passthrough, virtual attribute
+			"source_image_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 255),
```

Comment:
I believe this is because the read operation always returns a different value for `ImageId` than what is provided during the create operation. The `source_` prefix distinguishes the value passed to the Create API from what gets read back (and also explains why `image_id` is computed only).